cmd/kube-proxy/app/options.go

```diff
@@ -328,6 +315,50 @@ func (o *Options) processV1Alpha1Flags(fs *pflag.FlagSet) {
 	if fs.Changed("cluster-cidr") {
 		o.config.DetectLocal.ClusterCIDRs = strings.Split(o.clusterCIDRs, ",")
 	}
+	if fs.Changed("healthz-bind-address") {
+		host, port, err := net.SplitHostPort(o.healthzBindAddress)
+		var ip net.IP
+
+		// o.healthzBindAddress can have both "ip" and "ip:port" as value,
+		// we need to handle both cases.
+		if err != nil {
+			ip = netutils.ParseIPSloppy(o.healthzBindAddress)
+		} else {
+			ip = netutils.ParseIPSloppy(host)
+			intPort, _ := strconv.Atoi(port)
+			o.config.HealthzBindPort = int32(intPort)
+		}
+
+		if ip.IsUnspecified() {
+			o.config.HealthzBindAddresses = []string{fmt.Sprintf("%s/0", ip.String())}
+		} else if netutils.IsIPv4(ip) {
+			o.config.HealthzBindAddresses = []string{fmt.Sprintf("%s/32", ip.String())}
+		} else {
+			o.config.HealthzBindAddresses = []string{fmt.Sprintf("%s/128", ip.String())}
```

Comment:
I spent 30 mins to try to understand this PR and this is the source of my confusion, we calling these things addresses but we are using cidr notation to store it, so we can reuse the other existing methods ... this sounds super hacky